### PR TITLE
Add clone from VolumeSnapshot docs

### PR DESCRIPTION
**Pull request by akalenyu**
Signed-off-by: Alex Kalenyuk <akalenyu@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: doc/clone-from-volumesnapshot-source.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Data Volume cloning from a VolumeSnapshot source
+
+## Introduction
+Current CDI PVC cloning methods offer [smart cloning](./smart-clone.md) which uses a 1:1 cloning method that creates a temporary snapshot of the source PVC for each intended clone.  
+Some storage systems are designed to scale better with a 1:N method where a single snapshot of the source PVC is used to create multiple clones.  
+One examples is ceph, where this would be about utilizing RBD layering.  
+A much simpler example would be storage that can take around 5-10 minutes to snapshot a 200GB volume.  
+
+### Note: Data Volume cloning from snapshot source can work together with namespace transfer and size expansion
```
Comment (ShellyKa13) [inline doc/clone-from-volumesnapshot-source.md:9]: Note is usually marked with `>` or `*` not as another headline i.e ###

File: doc/clone-from-volumesnapshot-source.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  3) Storage supports expansion (if the user attempts clone to larger target)
+
+### Flow Description
+- DataVolume is created with a Snapshot source
+- Check if restore from said snapshot is possible (as described in Prerequisites):
+- If possible:
+    * Create the restore PVC
+    * Set the claim reference of the PV to point to the new target PVC
+- If not possible:
+    * Attempt [host-assisted cloning](./clone-datavolume.md) between 2 PVCs where CDI creates a temporary restore PVC from the snapshot to act as the source.
```
Comment (ShellyKa13) [inline doc/clone-from-volumesnapshot-source.md:23]: I'm not sure I understand, you write this Attempt under the `if not possible` which refers to if not possible to restore from said snapshot yet here you write that you create a temp restore PVC to do regular host assisted clone. what did I miss?

File: doc/clone-from-volumesnapshot-source.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+### Prerequisites
+  1) The source snapshot and target PVC share the same provisioner
+  2) The user creating the DataVolume has permission to create the `datavolumes/source` resource in the source namespace
+  3) Storage supports expansion (if the user attempts clone to larger target)
+
+### Flow Description
+- DataVolume is created with a Snapshot source
+- Check if restore from said snapshot is possible (as described in Prerequisites):
+- If possible:
+    * Create the restore PVC
```
Comment (ShellyKa13) [inline doc/clone-from-volumesnapshot-source.md:20]: maybe mentions that this is a temp PVC that will be deleted after?

File: doc/clone-from-volumesnapshot-source.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    * Set the claim reference of the PV to point to the new target PVC
```
Comment (ShellyKa13) [inline doc/clone-from-volumesnapshot-source.md:21]: maybe mention that you change the claim ref of the restore PV from the restore PVC to the target PVC?

File: doc/clone-from-volumesnapshot-source.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataVolume
+metadata:
+  name: golden-snap-source
+  namespace: golden-ns
+spec:
+  source:
+      http:
+         url: http://.../Fedora-Cloud-Base-34-1.2.x86_64.qcow2
```
Comment (ShellyKa13) [inline doc/clone-from-volumesnapshot-source.md:37]: consider using an actual path that the user can try and use?

**Review by ShellyKa13 — COMMENTED** [review]
Nice! had some suggestions and a question

File: doc/clone-from-volumesnapshot-source.md (hunk 1 above)
Comment (akalenyu) [reply doc/clone-from-volumesnapshot-source.md]: thx, done

File: doc/clone-from-volumesnapshot-source.md (hunk 2 above)
Comment (akalenyu) [reply doc/clone-from-volumesnapshot-source.md]: Yeah this is confusing, added `Check if restore from said snapshot to desired target is possible` to tackle
Consider someone cloning a target that has a different provisioner. We can still restore from the snapshot and attempt host assisted

The idea is that we provide a "complete" clone from volumesnapshot package

File: doc/clone-from-volumesnapshot-source.md (hunk 3 above)
Comment (akalenyu) [reply doc/clone-from-volumesnapshot-source.md:20]: Added

File: doc/clone-from-volumesnapshot-source.md (hunk 4 above)
Comment (akalenyu) [reply doc/clone-from-volumesnapshot-source.md]: I added a link to ns transfer doc

File: doc/clone-from-volumesnapshot-source.md (hunk 5 above)
Comment (akalenyu) [reply doc/clone-from-volumesnapshot-source.md]: Done

File: doc/clone-from-volumesnapshot-source.md (hunk 2 above)
Comment (ShellyKa13) [reply doc/clone-from-volumesnapshot-source.md]: ahh ok got it!

**Pull request merged**